refactor(app): Improve RunModuleInfo typing

SyntaxColoring opened the pull request:
## Overview

A little type-checking fixup, following up on https://github.com/Opentrons/opentrons/pull/19164#discussion_r2263362646.

## Test Plan and Hands on Testing

Just CI.

## Changelog

## Review requests

None.

## Risk assessment

Very low.

SyntaxColoring committed:
Improve getRunModuleRenderInfo() typing.

These props are nullable when they're passed to <Module> because <Module> needs to support the case where it's being rendered in isolation, not atop a deck. This util, on the other hand, needs to guarantee that it outputs non-null, because it *is* used for aligning modules to decks, and we need these props to compute the transforms to do that.

SyntaxColoring committed:
Update call sites.

smb2268 commented on app/src/organisms/ErrorRecoveryFlows/shared/TwoColLwInfoAndDeck.tsx:
```diff
@@ -131,18 +131,16 @@ export function TwoColLwInfoAndDeck(
-                  : null,
-            }
-          })
+        const modulesOnDeck = moduleRenderInfo.map(module => {
```
```suggestion
        const modulesOnDeck = moduleRenderInfo?.map(module => {
```
Should fix that failing test! Or you can add a mock so that moduleRenderInfo is not missing when the test runs

SyntaxColoring replied:
It seems like the mock is intentionally passing in `null`, to test that we handle "invalid props" correctly? Not sure why, but I'm restoring the `?.` and leaving the mock alone.

smb2268 reviewed (approved):
Ty!

SyntaxColoring committed:
OK, account for null moduleRenderInfo.

This shouldn't be necessary according to the type annotations, but the tests check for this explicitly, so...? :shrug:
